README.md

```diff
@@ -28,46 +28,94 @@ The zkEVM back-end is called `EraVM`, and the architecture is called `eravm`.
-   3.b. `cargo install compiler-llvm-builder --target x86_64-unknown-linux-musl` on Linux for distribution
+   * MacOS:
 
-   The builder is not the zkEVM LLVM framework itself, but a tool that clones its repository and runs the sequence of build commands.
-   By default it is installed in `~/.cargo/bin/`, which is recommended to be added to your `$PATH`.
+      * Install [HomeBrew](https://brew.sh) package manager.
+      * Install the following packages:
 
-4. In a directory in which you want the `llvm` directory, create an `LLVM.lock` file with the URL and branch or tag you want to build. For example:
+         ```shell
+         # Install the required packages using `brew`
+         brew install cmake ninja coreutils
+         ```
 
-  ```
-  url = "<THIS REPO URL>"
-  branch = "<THIS REPO BRANCH>"
-  ```
+      * Install your choice of a recent LLVM/[Clang](https://clang.llvm.org) compiler, e.g. via [Xcode](https://developer.apple.com/xcode/), [Apple’s Command Line Tools](https://developer.apple.com/library/archive/technotes/tn2339/_index.html), or your preferred package manager.
+</details>
 
-5. Run the builder to clone and build the zkevm LLVM framework:
-   5.1. `zkevm-llvm clone`
-   5.2. `zkevm-llvm build`
+<details>
+<summary>2. Install Rust.</summary>
 
-   The build artifacts will end up in the `./target-llvm/target-final/` directory.
-   You may point your `LLVM_SYS_150_PREFIX` to that directory to use this build as a compiler dependency.
-   If built with the `--enable-tests` option, test tools will be in the `./target-llvm/build-final/` directory, along with copies of the build artifacts.
+   * Follow the latest [official instructions]((https://www.rust-lang.org/tools/install)):
+      ```shell
+      curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+      . ${HOME}/.cargo/env
+      ```
+
+      > Currently we are not pinned to any specific version of Rust, so just install the latest stable build for your   platform.
+</details>
+
+<details>
+<summary>3. Install the zkEVM LLVM framework builder.</summary>
+
+   * Install the builder using `cargo`:
+      ```shell
+      # Download and install the builder using `cargo`
+      cargo install compiler-llvm-builder
+      ```
+
+      > The builder is not the zkEVM LLVM framework itself, but a tool that clones its repository and runs the sequence of build commands. By default it is installed in `~/.cargo/bin/`, which is recommended to be added to your `$PATH`.
+
+</details>
+
+<details>
+<summary>4. Create `LLVM.lock` file.</summary>
+
+   * In a directory in which you want the `llvm` directory, create an `LLVM.lock` file with the URL and branch or tag you want to build. For example:
+
+      ```properties
+      url = "https://github.com/matter-labs/era-compiler-llvm"
+      branch = "main"
+      ```
+
+</details>
+
+<details>
+<summary>5. Build LLVM.</summary>
+
+   * Clone and build the zkEVM LLVM framework using the `zkevm-llvm` tool:
```

Comment:
See previous comment on whether we want instruction+comment in code block, or just comment in code block.